egs/wsj/s5/steps/nnet3/train_dnn.py

```diff
@@ -298,13 +299,17 @@ def TrainNewModels(dir, iter, srand, num_jobs, num_archives_processed, num_archi
                                                # the other indexes from.
         archive_index = (k % num_archives) + 1 # work out the 1-based archive index.
         frame = (k / num_archives) % frames_per_eg
+        offset_num = -1
+        if num_cmn_offsets > 0:
+          offset_num = (archive_index - 1) % num_cmn_offsets
```

Comment:
Dan, I fixed this issue in my local repo as follows, but I didn't commit the change.
epoch_index = num_archives_processed / (num_archives * frames_per_eg)
offset_num = epoch_index % num_cmn_offsets

e.g. num_archives_to_process = num_epochs * frames_per_eg * num_archives =>
epoch_index = num_archives_processed / ( frames_per_eg * num_archives)

Isn't that what we expect?
Also I think we need to process the data and its perturbed versions randomly.
e.g. in ubm offset model, while adding offset in 2nd epoch, I got large log-likelihood degradation in the beginning of 2nd epoch.
Any idea?